drivers/display/display_sdl.c

```diff
@@ -509,34 +597,57 @@ static int sdl_display_clear(const struct device *dev)
 		return -EINVAL;
 	}
 	LOG_DBG("size: %zu, bgcolor: %hhu", size, bgcolor);
-	memset(disp_data->buf, bgcolor, size);
+
+	K_SPINLOCK(&disp_data->lock) {
+		memset(disp_data->buf, bgcolor, size);
+	}
 
 	return 0;
 }
 
 static int sdl_display_blanking_off(const struct device *dev)
 {
 	struct sdl_display_data *disp_data = dev->data;
+	struct sdl_display_task task = {
+		.op = SDL_BLANKING_OFF,
+	};
 
 	LOG_DBG("Turning display blacking off");
 
-	disp_data->display_on = true;
+	K_SPINLOCK(&disp_data->lock) {
+		disp_data->display_on = true;
+	}
```

Comment:
```suggestion
	disp_data->display_on = true;
```
In native_sim a SW thread cannot be interrupted while executing simple instructions like this.
Anyhow, setting a variable to 1/0 is atomic in any platform I know of :)
So this is just overhead.
Same in `sdl_display_blanking_on()`

Reply:
Done.